tests/test_parquet.py

```diff
@@ -642,6 +642,13 @@ def testWriteSingleIndexDataFrameReadAsArrowTable(self):
             if schema.field(name).type not in (pa.string(), pa.binary()):
                 self.assertEqual(schema.field(name).type, schema2.field(name).type)
 
+        # Check getting a component while overriding the storage class via
```

Comment:
```suggestion
        # Check getting a component while overriding the parent storage class via
```
This is an important distinction because what's happening is that you are changing from DataFrame to ArrowAstropy and then using columns which ends up with a different definition automatically, whereas above the test is using DataFrame.columns and then changing the storage class.

This code works already:
```python
columnsType = self.datasetType.makeComponentDatasetType("columns").overrideStorageClass("ArrowColumnList")
self.assertEqual(columns2, self.butler.get(columnsType))
```
but could you add that test as well to make it clear that the order of the component making / overriding matters?